CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.cc

```diff
@@ -376,6 +396,38 @@ void EcalTPGParamBuilder::analyze(const edm::Event& evt, const edm::EventSetup&
   evtSetup.get< EcalMappingRcd >().get(ecalmapping);
   theMapping_ = ecalmapping.product();
 
+  // get record for alpha
+  edm::LogInfo("Alphas Info : ") << "EcalLaserDbAnalyzer::analyze " << "\n";
+  edm::ESHandle<EcalLaserAlphas> handle;
+  evtSetup.get<EcalLaserAlphasRcd>().get(handle);
+  edm::LogInfo("Alphas Info : ") << "EcalLaserDbAnalyzer::analyze-> got EcalLaserDbRecord: " << "\n";
+  EcalLaserAlpha alpha;
+  const EcalLaserAlphaMap& laserAlphaMap = handle.product()->getMap(); // map of apdpns
+
+  //modif-alex-27-july-2015-beg
+  // use alpha to check
+  int cnt =0;
+  for(int ieta=-EBDetId::MAX_IETA; ieta<=EBDetId::MAX_IETA; ++ieta) {
+    if(ieta==0) continue;
+    for(int iphi=EBDetId::MIN_IPHI; iphi<=EBDetId::MAX_IPHI; ++iphi) {
+      EBDetId ebdetid(ieta,iphi);
+      EcalLaserAlphaMap::const_iterator italpha = laserAlphaMap.find( ebdetid );
```

Comment:
@depasse - it looks like this find inside a double loop is better implemented as a loop over laserAlphaMap (you can fill a map of ieta iphi if the check for missing laser conditions is important).